File: src/main/java/org/digma/intellij/plugin/jaegerui/model/outgoing/Insight.java
package org.digma.intellij.plugin.jaegerui.model.outgoing;


import com.fasterxml.jackson.annotation.*;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
@JsonSerialize
public record Insight(@JsonProperty(value = "type",required = true) String type,
                      @JsonProperty(value = "importance",required = true)int importance) {

    @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
    public Insight {
        //nothing to do , for jackson only
    }
}

File: src/main/kotlin/org/digma/intellij/plugin/ui/jaegerui/model/Insight.kt
package org.digma.intellij.plugin.ui.jaegerui.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.beans.ConstructorProperties


@JsonIgnoreProperties(ignoreUnknown = true)
data class Insight
@JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
@ConstructorProperties("type", "importance")
constructor(
    val type: String,
    val importance: Int
)